### ISSUE TITLE
[Awaiting C+ payment - $500] [HOLD for payment 2023-08-16] Implement the MapboxAccessTokenManager

### ISSUE THREAD

**Issue by neil-marcellini** [Expensify/App#22701]
Please follow the [plan in the design doc](https://docs.google.com/document/d/16Q9sE-991CZ4wUcalOs_U-qyEEza_C9tZwj6G5lHUp0/edit#heading=h.91s1gs72rev4). Be sure to write a manual or automated test to make sure it gets the token properly.
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012e60fd48447acbeb</li>
        <li>Upwork Job ID: 1707369473068421120</li>
        <li>Last Price Increase: 2023-09-28</li>
    </ul>
</details>

**Comment by luacmartins** [comment]
I can pick up this one since it's on hold for the issue I'm working on and is related to that code.

**Comment by tgolen** [comment]
I'm gonna start working on this since it looks like nothing has been done yet and Carlos is OOO today.

**Comment by tgolen** [comment]
still on hold

**Comment by tgolen** [comment]
still on hold while deploys happen

**Comment by neil-marcellini** [comment]
The API was deployed to staging so it's off hold now!

**Comment by tgolen** [comment]
OK, the PR is all written and just needs to be tested and reviewed. I'll get that done today.

**Comment by tgolen** [comment]
This was done internally so can go ahead and be closed.

**Comment by situchan** [comment]
@tgolen C+ review for internal PR is pending

**Comment by situchan** [comment]
> @tgolen C+ review for internal PR is pending

@tgolen bump for payment
Can you please add `NewFeature` label to assign BZ member?

**Comment by tgolen** [comment]
@michaelhaxhiu Could you please assist with settling up payment on this one? The C+ needs compensated for a review on the PR.

**Comment by michaelhaxhiu** [comment]
@sobitneupane let's get you paid

**Comment by michaelhaxhiu** [comment]
Based on timestamps, @sobitneupane should get a $1000 flat rate for C+ review.

You are paid via newDot, can you please request payment and link this comment to the request appropriately?

**Comment by michaelhaxhiu** [comment]
Let us know when it's done and we'll close this GH.

**Comment by michaelhaxhiu** [comment]
I'm re-assigning this to another BZ as part of my preparation for Sabbatical (starting Friday).

**Next steps:**
- Wait for @sobitneupane to request payment via NewDot 
- After that's confirmed, close this GH 🎬

**Comment by sophiepintoraetz** [comment]
FYI - reviews are $750, not $1000 @michaelhaxhiu

**Comment by situchan** [comment]
This work was done long time ago, before price drop announcement.
So original base price $1000 applies here.
PR: https://github.com/Expensify/App/pull/23511

**Comment by sophiepintoraetz** [comment]
Ahk, that context was missing. Thanks for the update @situchan!

**Comment by sobitneupane** [comment]
I think @situchan was the one who reviewed the PR, not me.

**Comment by sophiepintoraetz** [comment]
Thanks for the honesty and correcting Hax there! Waiting on @situchan to request $1000 on ND for the PR review. I'm going to go ahead and close this, seeing as that's the only outstanding action.

cc @JmillsExpensify

**Comment by situchan** [comment]
@sophiepintoraetz sorry, I am still using upwork

**Comment by sophiepintoraetz** [comment]
Okay, offer sent!
